orttraining/orttraining/training_ops/cuda/loss/softmax_cross_entropy_loss_impl.cu

```diff
@@ -117,7 +117,11 @@ __global__ void _WeightedSoftmaxCrossEntropyLossGrad(
   int row = i / C;
   int d = i % C;
   CUDA_KERNEL_ASSERT(weight[row] == 0 || (label[row] >= 0 && label[row] < C));
-  output_data[i] = (*dY) * weight[row] * (_Exp(log_prob[i]) - 1.0 * (d == label[row])) / (*normalize_factor);
+  if(0 == *normalize_factor){
+    output_data[i] = 0;
```

Comment:
Can you add a comment to explain?
```suggestion
    // normalize_factor is sum of labels' weights. Because zero 
    // means implies all weights are 0, the loss function should 
    // be constant 0 and its corresponding gradient should be 0 as well.
    output_data[i] = 0;
```

Reply:
Seems like your comment came in after the merge but I will add it in subsequent PR. Thanks

Reply:
Addressed at https://github.com/microsoft/onnxruntime/pull/3972